storage/remote/queue_manager.go

```diff
@@ -1021,6 +1022,10 @@ func (t *QueueManager) shouldReshard(desiredShards int) bool {
 		level.Warn(t.logger).Log("msg", "Skipping resharding, last successful send was beyond threshold", "lastSendTimestamp", lsts, "minSendTimestamp", minSendTimestamp)
```

Comment:
I considered whether it was possible to combine the logic for lastSendTimestamp with this new logic, but I wasn't able to identify a simple path. 

I'm also not sure if the older condition has value any more in favor of the new check: if we were to remove it, resharding would still be disabled as soon as the first request times out (as that would be a retriable error).

Reply:
I think I would keep both, or simplify to one somehow. It makes sense to _not_ reshard if we're running into recoverable or unrecoverable errors.